### DISCUSSION TITLE
LND / Neutrino it's not running RPC server.

### DISCUSSION THREAD

**Discussion by miguelkmldev** [lightningnetwork/lnd#6473] [Troubleshooting]
### Background

Describe your issue here.

### Your environment

* version of `lnd` : 0.14.2-beta.rc2
* which operating system :  iOS IPhone 13 simulator
* using Neutrino node

### Steps to reproduce

I have been working on a product 100% non centralized, I'm following this guide [Neutrino guide](https://lightning.engineering/posts/2019-11-15-mobile-lnd/) to setup node on mobile device.
The node its running but its not running the RPC server only the REST server.

The command that I'm uising ro run the node is 

`--bitcoin.active --bitcoin.testnet --bitcoin.node=neutrino --lnddir=\(path) --neutrino.connect=faucet.lightning.community --no-macaroons --debuglevel=debug --tlsdisableautofill --no-rest-tls --rpclisten=127.0.0.1:10009`

### Expected behaviour

![image](https://user-images.githubusercontent.com/100699772/164749607-56d7bab7-d869-48a3-9553-711c628d13f1.png)

### Actual behaviour
![image](https://user-images.githubusercontent.com/100699772/164749097-43b2fa3c-a6a6-4e45-9816-ef9b1446e468.png)

**Comment by tvolk131** [comment]
Could you paste a full version of your logs when you run the command?

**Comment by miguelkmldev** [comment]
Sure!

![image](https://user-images.githubusercontent.com/100699772/165104291-60b8ff9e-20b8-42a1-a4d6-4e6e9614d52b.png)

**Comment by guggero** [comment]
Please don't post screenshots of logs. That makes it so much harder to read through things... And I now have to type the log message I'm referencing instead of just being able to copy/paste.

```
RPC server listening on bufconn
```

That's normal for the mobile use case since all RPCs are run through the stub code generated for mobile.
If you want an _additional_ TCP listener, you need to edit the `mobile/bindings.go` file and add an extra `ListenerWithSignal` to the `lnd.ListenerCfg` on line 96/97.
But I guess that defeats the whole purpose of having the mobile stubs that connect in-memory. So not sure what you're trying to achieve?

**Comment by miguelkmldev** [comment]
@guggero sorry for the mistake pasting the screenshot with the logs.

Let me explain what's my idea.... I'm trying to run LND on mobile as mentioned before, and expect to communicate with the node capabilites using gRPC or REST API  using flutter app as client, but without RPC server up I can't do nothing. 

Following this guide [Neutrino Guide](https://lightning.engineering/posts/2019-11-15-mobile-lnd/) I could to generate the Lndmobile library to use on IOS / Android, and generate the gRPC methods from the .proto files, but I had some issues generating the code base for Java (Android) for this reason decide to generate .dart files from .proto files and manage only one codebase to communicate with the node capabilities, but in this step I found the problem with the RPC server, trying to hit the rest API or with gRPC generated files a UNAVAILABLE error it's present.

**Comment by guggero** [comment]
I guess that use case makes sense. But because of the way the mobile bindings currently work, you'll need to edit the `mobile/bindings.go` file in `lnd` as mentioned above to add an extra listener, then generate the stub code again and import into your project to enable the gRPC server.

**Comment by miguelkmldev** [comment]
@guggero thank you for your help.

I'll be working on it, if suddenly you have any guide to follow I'll very grateful if you can share it.

**Comment by hsjoberg** [comment]
Hi @miguelkmldev, I would advice against trying to spin up up real gRPC server. There's a history behind the design decisions that lead up to the current mobile bindings. https://www.youtube.com/watch?v=bjZ_QRgSAJ8

You can check out my code on how I got protobuf codegen to work on Android here: 
https://github.com/hsjoberg/blixt-wallet/blob/a5b03733573c89eae7bbd4a24f38ec61e0bdfa81/android/app/build.gradle
https://github.com/hsjoberg/blixt-wallet/blob/a5b03733573c89eae7bbd4a24f38ec61e0bdfa81/android/build.gradle#L17

I also have fully developed lnd gluecode for iOS/macOS and Android there as well.

Back when I started I went with the "gRPC server on device" route, because the current in-memoery gRPC idea had not been conceived back then.
But it led to clumsy and slow startup times and in general an unsatisfying result.

**Comment by miguelkmldev** [comment]
Hi @hsjoberg at the beginning I have been working with this focus, generate de bindings for IOS / Android, but as mentioned before, I had some issues in java, I can generate the methods but specifically one of this files was generated with errors, and it's required by others modules then other required files are broken.

In the case of `lnrpc/LigthningOuterClass.java` it's not generated as Java Class, and `lnrpc/WalletUnlocker.java` try to use it.

![image](https://user-images.githubusercontent.com/100699772/165185825-c33463d5-6c04-4004-ad1e-7dcb02975fda.png)

Please let me know if this error it's known or if should create a new issue related to that.

**Comment by Roasbeef** [comment]
I'm gonna convert this to a Discussion since there's no issue w/ lnd, and instead the OP is attempting to embed the gRPC server instead of use the mobile bindings. 

You may also be interested in this PR @miguelkmldev: https://github.com/lightningnetwork/lnd/pull/6464
